crates/ruma-client-api/src/push.rs

```diff
@@ -110,6 +111,25 @@ impl From<PatternedPushRuleInit> for PushRule {
     }
 }
 
+impl From<AnyPushRule> for PushRule {
+    fn from(push_rule: AnyPushRule) -> Self {
+        match push_rule {
+            AnyPushRule::Override(r) => r.into(),
+            AnyPushRule::Content(r) => r.into(),
+            AnyPushRule::Room(r) => r.into(),
+            AnyPushRule::Sender(r) => r.into(),
+            AnyPushRule::Underride(r) => r.into(),
+            _ => unreachable!(),
```

Comment:
I'm not 100% sure why the checks fail because of this. I'm not sure what else to use, since the types are in different crates but we know there are only these variants.

Reply:
This is because with `--all-features`, we activate `unstable-exhaustive-types`, and `AnyPushRule` becomes exhaustive. I think the best solution is to `#[allow]` the lint here with a comment.

Reply:
That makes the feature non-additive, let's not do that.